File: app/src/main/java/com/nowfloats/Product_Gallery/Model/Product.java
package com.nowfloats.Product_Gallery.Model;

import android.net.Uri;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

public class Product implements Serializable {

    //public String BuyOnlineLink;
    @SerializedName(value="currencyCode", alternate={"CurrencyCode"})
    public String CurrencyCode;
    @SerializedName(value="description", alternate={"Description"})
    public String Description;
    @SerializedName(value="discountAmount", alternate={"DiscountAmount"})
    public double DiscountAmount;
    public String ExternalSourceId;
    public String IsArchived;
    @SerializedName(value="isAvailable", alternate={"IsAvailable"})
    public boolean IsAvailable;
    @SerializedName(value="isFreeShipmentAvailable", alternate={"IsFreeShipmentAvailable"})
    public String IsFreeShipmentAvailable;
    @SerializedName(value="name", alternate={"Name"})
    public String Name;
    @SerializedName(value="price", alternate={"Price"})
    public double Price;
    @SerializedName(value="priority", alternate={"Priority"})
    public String Priority;
    public String ShipmentDuration;
    public int availableUnits;
    public ArrayList<String> _keywords;
    public String ApplicationId;
    @SerializedName(value="fPTag", alternate={"TPTag"})
    public String FPTag;
    @SerializedName(value="clientId", alternate={"ClientId"})
    public String ClientId;

    public String ImageUri;
    public String ProductUrl;
    public ArrayList<ImageListModel> Images;
    public String MerchantName;
    public String TileImageUri;
    public String _id;
    public String GPId;
    public String TotalQueries;
    public String CreatedOn;
    public String ProductIndex;
    public Uri picimageURI =null;
    public String UpdatedOn;
    public boolean isProductSelected;

    public String productType;
    public String paymentType;
    public boolean variants;
    public String brandName;
    public String category;

    @SerializedName("isCodAvailable")
    public boolean codAvailable;
    @SerializedName("maxCodOrders")
    public int maxCodOrders;
    @SerializedName("isPrepaidOnlineAvailable")
    public boolean prepaidOnlineAvailable;
    @SerializedName("maxPrepaidOnlineOrders")
    public int maxPrepaidOnlineAvailable;

    public BuyOnlineLink BuyOnlineLink;
    public Specification keySpecification;
    public List<Specification> otherSpecification;


    public static class BuyOnlineLink implements Serializable
    {
        public String url;
        public String description;
    }

    public static class Specification implements Serializable
    {
        public String key;
        public String value;
    }
}

File: appService/src/main/java/com/appservice/model/serviceProduct/Product.kt
package com.appservice.model.serviceProduct

import android.net.Uri
import com.appservice.model.KeySpecification
import com.google.gson.annotations.SerializedName
import java.io.Serializable
import java.util.*

class Product(
    @SerializedName(value = "currencycode", alternate = ["CurrencyCode"])
    var CurrencyCode: String? = null,

    @SerializedName(value = "description", alternate = ["Description"])
    var Description: String? = null,

    @SerializedName(value = "discountAmount", alternate = ["DiscountAmount"])
    var DiscountAmount: Double = 0.0,
    var ExternalSourceId: String? = null,
    var IsArchived: String? = null,

    @SerializedName(value = "isAvailable", alternate = ["IsAvailable"])
    var IsAvailable: Boolean = false,

    @SerializedName(value = "isFreeShipmentAvailable", alternate = ["IsFreeShipmentAvailable"])
    var IsFreeShipmentAvailable: String? = null,

    @SerializedName(value = "name", alternate = ["Name"])
    var Name: String? = null,

    @SerializedName(value = "price", alternate = ["Price"])
    var Price: Double = 0.0,

    @SerializedName(value = "priority", alternate = ["Priority"])
    var Priority: String? = null,
    var ShipmentDuration: String? = null,
    var availableUnits: Int = 1,
    var _keywords: ArrayList<String>? = null,
    var tags: ArrayList<String>? = null,
    var ApplicationId: String? = null,

    @SerializedName(value = "fpTag", alternate = ["FPTag"])
    var FPTag: String? = null,

    @SerializedName(value = "clientId", alternate = ["ClientId"])
    var ClientId: String? = null,
    var ImageUri: String? = null,
    var ProductUrl: String? = null,
    var Images: ArrayList<ImageListModel>? = null,
    var MerchantName: String? = null,
    var TileImageUri: String? = null,

    @SerializedName("_id")
    var productId: String? = null,
    var GPId: String? = null,
    var TotalQueries: String? = null,
    var CreatedOn: String? = null,
    var ProductIndex: String? = null,
    var picimageURI: Uri? = null,
    var UpdatedOn: String? = null,
    var isProductSelected: Boolean = false,
    var productType: String? = null,
    var paymentType: String? = null,
    var variants: Boolean = false,
    var brandName: String? = null,
    var category: String? = null,

    @SerializedName("isCodAvailable")
    var codAvailable: Boolean = false,

    @SerializedName("maxCodOrders")
    var maxCodOrders: Int = 10,

    @SerializedName("isPrepaidOnlineAvailable")
    var prepaidOnlineAvailable: Boolean = true,

    @SerializedName("maxPrepaidOnlineOrders")
    var maxPrepaidOnlineAvailable: Int = 10,

    @SerializedName("uniquePaymentUrl")
    var BuyOnlineLink: BuyOnlineLink? = null,
    var keySpecification: KeySpecification? = null,

    @SerializedName("otherSpecifications")
    var otherSpecification: ArrayList<KeySpecification>? = null,
    var pickupAddressReferenceId: String? = null
) : Serializable {

  enum class PaymentType(val value: String) {
    ASSURED_PURCHASE("AssuredPurchase"), MY_PAYMENT_GATEWAY("MyPaymentGateWay"), UNIQUE_PAYMENT_URL("UniquePaymentUrl"), DONT_WANT_TO_SELL("None");

    companion object {
      fun fromValue(value: String): PaymentType? = values().firstOrNull { it.value.toLowerCase(Locale.ROOT) == value.toLowerCase(Locale.ROOT) }
    }
  }
}